build-system/server.js

```diff
@@ -33,6 +33,14 @@ var url = require('url');
 app.use(bodyParser.json());
 app.use(morgan('dev'));
 
+app.use('/', function(req, res, next) {
+  res.setHeader('AMP-Access-Control-Allow-Source-Origin',
```

Comment:
Ok, this sounds like a reasonable approach.  Restructured the `server.js` changes:
  - Added a handler function for CORS requests that checks for the `__amp_source_origin` URL param and sets the response header only if present.
  - Restricted paths for CORS handling to `/examples/*` and `/extensions/*`.

On my workstation, at least, it passes `gulp test` and `gulp test --integration`.  Not sure what to do with manual tests b/c I'm not sure which (if any) use CORS.  The handful I looked at *appeared* to do reasonable things, but I'm also not super-sure what I should be looking for.  If you know of specific ones to test, let me know and I'll be happy to look at them.

Thanks!

PTAL